src/pages/PlaceAddModal.tsx

```diff
@@ -82,18 +105,29 @@ const PlaceAddModal = ({
         terms: [data.term],
       };
 
-      const resp = await addPlaceMutation.mutateAsync(formData);
-
-      if (!resp?.placeId) return;
-
-      const newPlace = await getPlaceById({ headers, id: resp.placeId });
-      onConfirmSuccess(newPlace);
+      try {
+        setDuplicatePlaceResponse(undefined);
+
+        const resp = await addPlaceMutation.mutateAsync(formData);
+
+        if (!resp?.placeId) return;
+        const newPlace = await getPlaceById({ headers, id: resp.placeId });
+        onConfirmSuccess(newPlace);
+      } catch (error) {
+        if (error?.status === DUPLICATE_STATUS_CODE) {
+          const body = error?.body;
```

Comment:
I would use the `DuplicatePlaceErrorBody` here to cast error.body to that type, since we know what the shape will be.

I would replace the `DuplicatePlaceErrorBody | any ` type in `FetchError` by just any, because it will go to any's value anyways

<img width="647" alt="Screenshot 2024-02-29 at 09 06 28" src="https://github.com/cultuurnet/udb3-frontend/assets/66943525/c5e02fdb-da2d-488e-95eb-1215c6a4ebb2">